File: client_front/src/app/Service/operation.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Operation } from '../model/operation';

@Injectable({
  providedIn: 'root'
})
export class OperationService {

   _url = "http://localhost:8081/operation/"
  // _urlClient = "http://localhost:8082/operation/"
  _urlClient="https://afoih-client.herokuapp.com/operation/"
 
  constructor(private http: HttpClient) { }



  getOperation(rib: string){
    return this.http.get<Operation>(this._urlClient+"listOperation/"+rib)
  }

  verser(operation: Operation){
    return this.http.post<any>(this._urlClient+"versement", operation)
  }

  retirer(operation: Operation){
    return this.http.post<any>(this._urlClient+"retrait", operation)
  }

  virer(operation: Operation){
    return this.http.post<any>(this._urlClient+"virement", operation)
  }

  recharge(operation: Operation, codeRacharge:number){
    return this.http.post<Operation>(this._urlClient+"recharge/"+codeRacharge, operation)
  }

  chercherOperation(motCle: String,rib:String){
    console.log(motCle);
    console.log(rib);
   return this.http.get(this._urlClient+"chercheO?mc="+motCle+"&rib="+rib);
    }

    cherche(motCle: String,rib:String){
      console.log(motCle);
      console.log(rib);
     return this.http.get(this._urlClient+"cherche/"+motCle+"/"+rib);
      }

  
}

File: client_back/src/main/java/com/ensa/controller/OperationService.java
package com.ensa.controller;

import com.ensa.e_banking.client.security.SecurityConstants;
import com.ensa.entities.Compte;
import com.ensa.entities.Operation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.ParameterizedTypeReference;
import org.springframework.http.*;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.client.RestTemplate;

import javax.servlet.http.HttpServletRequest;
import java.util.List;

@RestController
@CrossOrigin("*")
public class OperationService {
	@Autowired
	RestTemplate restTemplate;

	//private String url = "http://localhost:8081";
	private  String url="https://afoihi-agent.herokuapp.com/";
	HttpHeaders headers=new HttpHeaders();

	@RequestMapping(value="/operation/virement",method= RequestMethod.POST)
	@Transactional
	 public boolean virement(@RequestBody Operation operation,HttpServletRequest request) {

		headers.set(SecurityConstants.HEADER_STRING,
				SecurityConstants.TOKEN_PREFIX+request.getHeader(SecurityConstants.HEADER_STRING));
		HttpEntity<Operation> requ = new HttpEntity<>(headers);
		//Compte compteDestination = restTemplate.getForObject(url+"/compte/CC/rib/"+operation.getCompteDestination().getRib(), Compte.class);
		ResponseEntity<Compte> response = restTemplate.exchange(
				url+"/compte/CC/rib/"+operation.getCompteDestination().getRib(), HttpMethod.GET, requ, new ParameterizedTypeReference<Compte>() {}
		);
		Compte compteDestination= response.getBody();


		operation.setCompteDestination(compteDestination);
		headers.set(SecurityConstants.HEADER_STRING,
				SecurityConstants.TOKEN_PREFIX+request.getHeader(SecurityConstants.HEADER_STRING));
		HttpEntity<Operation> req = new HttpEntity<>(operation,headers);
		restTemplate.postForObject(url+"/operation/virement", req, boolean.class);
		return true;
	}

	@RequestMapping(value="/operation/recharge/{codeRecharge}",method= RequestMethod.POST)
	@Transactional
	 public boolean recharge(@RequestBody Operation operation, @PathVariable Long codeRecharge,HttpServletRequest request) {
		System.out.println("Recharge : "+operation.toString());
		headers.set(SecurityConstants.HEADER_STRING,
				SecurityConstants.TOKEN_PREFIX+request.getHeader(SecurityConstants.HEADER_STRING));
		HttpEntity<Operation> req = new HttpEntity<>(operation,headers);

		restTemplate.postForEntity(url+"/operation/recharge/"+codeRecharge, req, boolean.class);
		return true;
	}

	@RequestMapping(value="/operation/listOperation/{rib}",method= RequestMethod.GET)
	public List<Operation> getList(@PathVariable String rib, HttpServletRequest request){
		headers.set(SecurityConstants.HEADER_STRING,
				SecurityConstants.TOKEN_PREFIX+request.getHeader(SecurityConstants.HEADER_STRING));
		HttpEntity<Operation> entity = new HttpEntity<Operation>(headers);
		ResponseEntity<List<Operation>> response = restTemplate.exchange(
				url+"/operation/listOperation/"+rib, HttpMethod.GET, entity, new ParameterizedTypeReference<List<Operation>>() {}
		);
		List<Operation> list = response.getBody();
	return  list;
	}


	@RequestMapping(method=RequestMethod.GET, value="/operation/cherche/{mc}/{rib}")
	public List<Operation> chercheOperation (@PathVariable(name = "mc") String mc, @PathVariable(name = "rib") String rib,HttpServletRequest request){
		headers.set(SecurityConstants.HEADER_STRING,
				SecurityConstants.TOKEN_PREFIX+request.getHeader(SecurityConstants.HEADER_STRING));
		HttpEntity<Operation> entity = new HttpEntity<>(headers);

		ResponseEntity<List<Operation>> response = restTemplate.exchange(
				url+"/operation/chercheO/"+mc+"/"+rib, HttpMethod.GET, entity, new ParameterizedTypeReference<List<Operation>>() {}
		);

		List<Operation> list = response.getBody();
		return  list;
	}



}
